### DISCUSSION TITLE
[gatsby-source-contentful] Unavailable to dynamically change content?

### DISCUSSION THREAD

**Discussion by mwskwong** [gatsbyjs/gatsby#33496] [Help] [answered]
One of the advantages of using a headless CMS is the ability to change content without the need of recompiling/republishing the website. 

Given how `gatsby-source-contentful` works, does the plugin remove this ability?

**Comment by KyleAMathews** [comment, marked as answer]
Every possible way of publishing a website requires republishing part of it when you change content. What matters is how long it takes for the change to go live. For most Gatsby/Contentful websites, a change in Contentful is published to the website in < 10s. See the demo at the start of this blog post https://www.gatsbyjs.com/blog/2020-04-22-announcing-incremental-builds/
